src/snowflake/snowpark/_internal/compiler/plan_compiler.py

```diff
@@ -84,8 +94,31 @@ def compile(self) -> Dict[PlanQueryType, List[Query]]:
                 )
                 logical_plans = large_query_breakdown.apply()
 
+            after_complexities = [
```

Comment:
let's also add one after cte optimization. if cte is not enabled, it should be the same as original